crates/sui/src/ptb/ptb_parser/build_ptb.rs

```diff
@@ -647,54 +638,45 @@ impl<'a> PTBBuilder<'a> {
             }
             CommandToken::PickGasBudget => {
                 let PTBArg::Identifier(i) = command.args.pop().unwrap() else {
-                    bail!(
-                        "expected identifie at index {} in {}",
-                        self.command_index,
-                        self.current_context_name()
-                    );
+                    error!(self, "expected identifie",);
                 };
                 let picker = match i.to_string().as_str() {
                     "max" => GasPicker::Max,
                     "min" => GasPicker::Min,
                     "sum" => GasPicker::Sum,
-                    x => bail!(
-                        "invalid gas picker: {} at index {} in {}",
-                        x,
-                        self.command_index,
-                        self.current_context_name()
-                    ),
+                    x => error!(self, "invalid gas picker: {}", x,),
                 };
-                self.gas_budget.set_gas_picker(picker)?;
+                self.gas_budget
+                    .set_gas_picker(picker)
+                    .map_err(|e| err!(self, "{e}"))?;
```

Comment:
Is this error message sufficient?

Reply:
Yep! The actual interesting error is coming from teh gas picker, and all this is doing is adding the "command information" to it (i.e., command index)